pandas/core/arrays/base.py

```diff
@@ -515,9 +517,17 @@ def nbytes(self) -> int:
     # Additional Methods
     # ------------------------------------------------------------------------
 
-    def astype(self, dtype, copy=True):
+    @overload
+    def astype(self, dtype: NpDtype, copy: bool = ...) -> np.ndarray:
+        ...
+
+    @overload
+    def astype(self, dtype: Dtype, copy: bool = ...) -> ArrayLike:
```

Comment:
if we pass a EA dtype object, we get a union return type?

Reply:
Good point.  Fixed in next commit